File: Práctica/2.5 - RobocodeLabo/src/CasseraBottaBianchi/LaboRobot.java
package CasseraBottaBianchi;
import robocode.*;

public class LaboRobot extends JuniorRobot
{
	private RobotStrategy strategy = new EstrategiaGanadora(this);

	public void setStrategy(RobotStrategy strategy) {
		this.strategy = strategy;
	}
	@Override
	public void run() {
		setColors(orange, blue, white, yellow, black);
		this.strategy.run();
	}

	/**
	 * onScannedRobot: What to do when you see another robot
	 */
	@Override
	public void onScannedRobot() {
		this.strategy.onScannedRobot();
	}

	/**
	 * onHitByBullet: What to do when you're hit by a bullet
	 */
	@Override
	public void onHitByBullet() {
		this.strategy.onHitByBullet();
	}
	
	/**
	 * onHitWall: What to do when you hit a wall
	 */
	@Override
	public void onHitWall() {
		this.strategy.onHitWall();
	}	
}

File: Talleres/3/src/LaboRobot.kt
package CasseraBottaBianchi

import robocode.*

class LaboRobot : JuniorRobot() {

    private var strategy: RobotStrategy = EstrategiaGanadora(this)

    fun setStrategy(strategy: RobotStrategy) {
        this.strategy = strategy
    }

    override fun run() {
        setColors(orange, blue, white, yellow, black)
        strategy.run()
    }

    override fun onScannedRobot() {
        strategy.onScannedRobot()
    }

    override fun onHitByBullet() {
        strategy.onHitByBullet()
    }

    override fun onHitWall() {
        strategy.onHitWall()
    }
}
